### PR TITLE
add PlantDetail component

### LINKED ISSUE #12: build plant detail component
As a user I want to see my plant in the habit detail.

image

-------
- [x] plant
- [x] background 
- [x] stripe with dynamic habit color
- [x] text for each state
- [x] add stories for each state and different colors

### PR DESCRIPTION
**Pull request by jujuzitroni**
added the plant detail for the detail page.

also refactored the rating functionality, so that the rating can be accessed everywhere it is needed.

https://plants-need-plantdetail-piyj1u.herokuapp.com/storybook/?path=/story/component-plantdetail--final-stage

fixes #12

### CODE REVIEW

File: src/app/components/HabitPlant/HabitPlant.stories.tsx [hunk 1]
```diff
@@ -6,10 +6,16 @@ export default {
   component: HabitPlant,
 };
 
-export const NewlyPlanted = (): JSX.Element => <HabitPlant rating={0} />;
+export const NewlyPlanted = (): JSX.Element => (
+  <HabitPlant rating="empty_pot" />
```
Comment (lmachens) [inline src/app/components/HabitPlant/HabitPlant.stories.tsx:10]: `rating="empty_pot"`  has a different naming convention like you had before. 
What about `rating="emptyPot"`?
I would expect that a rating is a number or a string like "perfect", "beginner", "professional", but I guess it is similar to your approach.

File: src/app/components/HabitPlant/HabitPlant.tsx [hunk 2]
```diff
@@ -5,23 +5,23 @@ import plant_bloom from '../../assets/plant_bloom.png';
 import empty_pot from '../../assets/empty_pot.png';
 
 type HabitPlantProps = {
-  rating: number;
+  rating: string;
```
Comment (lmachens) [inline src/app/components/HabitPlant/HabitPlant.tsx:8]: There are only specific ratings possible.
Take a look at unions:
https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types

```suggestion
  rating: 'emptyPot' | 'freshPlant' | 'comingAlong';
```

You need this in `getImgSrc` and `getRating` too -> you can create a type for this:

```ts
type Rating = 'emptyPot' | 'freshPlant' | 'comingAlong';
```

File: src/app/components/PlantDetail/PlantDetail.module.css [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.plantCard {
+  display: grid;
+  justify-content: center;
+  justify-items: center;
```
Comment (lmachens) [inline src/app/components/PlantDetail/PlantDetail.module.css:4]: Are you sure you need both?

**Review by lmachens — COMMENTED** [review]
gj

**Review by dhnazarenko — COMMENTED** [review]
LGTM 🚀

File: src/app/components/HabitPlant/HabitPlant.tsx (hunk 2 above)
Comment (jujuzitroni) [reply src/app/components/HabitPlant/HabitPlant.tsx]: @lmachens I've moved the rating into the config file and added a rating type (see last commit). What do you think?

File: src/app/components/PlantDetail/PlantDetail.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  return 'great job';
+}
+
+function PlantDetail({ rating, color }: PlantDetailProps): JSX.Element {
+  const tagline = getTagline(rating);
+
+  return (
+    <article
+      className={styles.plantCard}
+      style={{ borderBottom: `8px solid ${color}` }}
```
Comment (MaxMuecket) [inline src/app/components/PlantDetail/PlantDetail.tsx:30]: Why you use inline style?

**Review by MaxMuecket — APPROVED** [review]
It looks absolutely stunning and cute. I like it a lot! ✌️

File: src/app/components/HabitPlant/HabitPlant.tsx (hunk 2 above)
Comment (lmachens) [reply src/app/components/HabitPlant/HabitPlant.tsx]: It's awesome φ(゜▽゜*)♪

**Pull request merged**